### DISCUSSION TITLE
Unable to view directly in browser CBR & epub files

### DISCUSSION THREAD

**Discussion by thenamelessthing** [devnoname120/epubviewer#13] [General]
I'm on the latest version of NC 27.1.9 with the plugin version 1.6.2 and when I click on a .cbr or .epub file nothing happened.

I tried with Edge, Chrome and Firefox

PDF seem to work properly.  Any suggestions or comments about this behavior?

https://i.imgur.com/sNri0Yg.gif

Thank you

**Comment by ptiloo2002** [comment]
I'm also with the  latest version of NC 27.1.9 with the plugin version 1.6.2.

When i click on a .cbr or .epub file, nothing happened too.

**Comment by smarinier** [comment]
Hi @devnoname120, there's an issue with the 

9eab5ba9c5f86572beb847ae8957f3cd0f439586.
    Viewer — Fix escaping of special symbols in the download URL

In main.ts, you are using encodedSource

-    downloadUrl = getAbsolutePath(file.source);
+    downloadUrl = getAbsolutePath(file.encodedSource);

But this is not available on NC 27, as @nextcloud/files are limited to 3.0.0 beta8. And encodedSource was added in the final 3.0.0 (just for NC 28 i guess). So this commit breaks the the EpubViewer for NC 27.

As issues are not available on this project (where is https://github.com/devnoname120/epubviewer/issues/5 ?) , i can't see why there was a need for this.encodedSource, as it works (for me) with this.source.

Yours,

**Reply by devnoname120** [reply]
> i can't see why there was a need for this.encodedSource, as it works (for me) with this.source.

See the release notes of https://github.com/devnoname120/epubviewer/releases/tag/1.6.2

See also:
- https://github.com/nextcloud/server/pull/42281
- https://github.com/nextcloud/server/issues/42171


> But this is not available on NC 27, as @nextcloud/files are limited to 3.0.0 beta8. And encodedSource was added in the final 3.0.0 (just for NC 28 i guess). So this commit breaks the the EpubViewer for NC 27.

Thanks for the heads-up. Could you do a PR? For example an if/else based on the Nextcloud version or the existence of `.encodedSource`.

**Reply by smarinier** [reply]
Hi, Done with PR #20

**Comment by thenamelessthing** [comment]
With 1.6.3 epub now work. thank you.


Now only CBR left to be fixed.

**Reply by devnoname120** [reply]
Feel free to make a pull request.
